### PR TITLE
Add types for react-star-ratings

### PR DESCRIPTION
**Pull request by vanillkay**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-star-ratings/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Type definitions for react-star-ratings 2.3
+// Project: https://github.com/ekeric13/react-star-ratings
+// Definitions by: Andrew Shkurenko <https://github.com/vanillkay>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
```
Comment (peterblazejewicz) [inline types/react-star-ratings/index.d.ts:5]:
```suggestion
```
it's probably from other package, minimum version tested/supported now is 3.7. No need to hardcode version unless specific reason (e.g TS feature)

File: types/react-star-ratings/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import * as React from "react";
+
+interface StarRatingProps {
+    rating?: number;
```
Comment (peterblazejewicz) [inline types/react-star-ratings/index.d.ts:10]:
```suggestion
    rating: number;
```
is this one required? https://github.com/ekeric13/react-star-ratings/blob/07add0e61bf77fb94a9613587e31bffa9f46364e/src/star-ratings.js#L180

Comment (peterblazejewicz) [reply types/react-star-ratings/index.d.ts:10]: same for some others

Comment (peterblazejewicz) [reply types/react-star-ratings/index.d.ts:10]: never mind, read the defualts

File: types/react-star-ratings/react-star-ratings-tests.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import StarRatings from 'react-star-ratings';
```
Comment (sandersn) [inline types/react-star-ratings/react-star-ratings-tests.tsx:1]: Use the commonjs syntax `import StarRatings = require('react-star-ratings')`.

Consumers of the package who want to use `import StarRatings from 'react-star-ratings'` will need to add `"esModuleInterop": true`, but Definitely Typed tests need to be backward compatible to show that the types will work with old code bases.

File: types/react-star-ratings/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "esModuleInterop": true,
```
Comment (sandersn) [inline types/react-star-ratings/tsconfig.json:7]: Please remove this to show that this package will work with old code bases that haven't turned it on yet.

File: types/react-star-ratings/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Installation
```
Comment (sandersn) [inline types/react-star-ratings/README.md:1]: The `@types` publisher generates a README. You do not need to write one yourself.

**Review by sandersn — CHANGES_REQUESTED** [review]

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@vanillkay thx!

**Review by sandersn — APPROVED** [review]

**Pull request merged**